tests/login.test.js

```diff
@@ -0,0 +1,97 @@
+/*global describe*/
+/*global test*/
+/*eslint-env node*/
+const app = require('../app.js');
+const request = require('supertest');
+var cookie;
+
+describe('POST /login/', function() {
+	test('POST /login/ respond with json', function(done) {
+		request(app)
+			.post('/login/')
+			.send({'username':'Screwee','password':'Password01'})
+			.set('Content-Type', 'application/json')
+			.set('Accept', 'application/json')
+			.expect(200)
+			.expect('Hello user Screwee')
+			.end(function(err,res) {
+				if (err) return done(err);
+				cookie = res.headers['set-cookie'];
+				done();
+			});
+	});
+});
+
+describe('GET /profile/', function() {
+	request(app)
+		.post('/login/')
+		.send({'username':'Screwee','password':'Password01'})
+		.set('Content-Type', 'application/json')
+		.set('Accept', 'application/json')
+		.end(function(err,res) {
+			cookie = res.headers['set-cookie'];
```

Comment:
and actually, since you don't really care about actual cookies, but just need to keep session, there's a wrapper around the supertest which does that already: https://www.npmjs.com/package/supertest-session

Reply:
I have added test-helpers file with function to login and pass authentication.
Plus I have added setting for items database so tests are now more independent.